provider/options.go

```diff
@@ -83,7 +85,10 @@ func getOpts(opts []Option) (config, error) {
 
 	// Validate config
 	if len(cfg.peerid) == 0 {
```

Comment:
Do we need to allow a peer ID to be configured without a host? Or, should we require that a host is provided instead? Then, if we always require a host, it should probably not be an option.

Reply:
Would it be better to require the host only, so that there is no guessing about when when to supply one or the other?

Reply:
If possible, I try to limit libp2p dependencies as much as possible on the `provider` to keep it modular, so that if we need to build a HTTP-only DHT provider in the future, we can reuse most of the existing code.

Having a mandatory `host` would allow to build more dependency on the libp2p host (which is currently optional).

Failing to provide a peerid (with either `WithPeerID()` or `WithHost()`) results in an error when building the `SweepingProvider` which fails early, with an explanation of how to address the issue.